src/main/java/org/prebid/server/bidder/smaato/SmaatoBidder.java

```diff
@@ -218,31 +224,29 @@ private User modifyUser(User user) {
             return Result.empty();
         }
 
-        final List<BidderError> errors = new ArrayList<>();
-        final List<BidderBid> bidderBids = bidResponse.getSeatbid().stream()
+        return Result.withValues(bidResponse.getSeatbid().stream()
                 .filter(Objects::nonNull)
                 .map(SeatBid::getBid)
                 .filter(Objects::nonNull)
                 .flatMap(Collection::stream)
-                .map(bid -> bidderBid(bid, bidResponse.getCur(), headers, errors))
+                .map(bid -> bidderBid(bid, bidResponse.getCur(), headers))
                 .filter(Objects::nonNull)
-                .collect(Collectors.toList());
-        return Result.of(bidderBids, errors);
+                .collect(Collectors.toList()));
     }
 
-    private BidderBid bidderBid(Bid bid, String currency, MultiMap headers, List<BidderError> errors) {
-        try {
-            final String markupType = getAdMarkupType(headers, bid.getAdm());
-            final Bid updateBid = bid.toBuilder().adm(renderAdMarkup(markupType, bid.getAdm())).build();
-            return BidderBid.of(updateBid, getBidType(markupType), currency);
-        } catch (PreBidException e) {
-            errors.add(BidderError.badInput(e.getMessage()));
-            return null;
+    private BidderBid bidderBid(Bid bid, String currency, MultiMap headers) {
+        final String bidAdm = bid.getAdm();
+        if (StringUtils.isBlank(bidAdm)) {
+            throw new PreBidException("Empty ad markup");
```

Comment:
Minor. It can be multiple bids, seems it will be helpful to know id of bidder which cause error